Modules/_sqlite/statement.c

```diff
@@ -48,13 +52,59 @@ typedef enum {
     TYPE_UNKNOWN
 } parameter_type;
 
+int pysqlite_statement_is_dml(sqlite3_stmt *st, const char *sql)
+{
+    const char* p;
+    int is_dml = 0;
+
+#ifdef HAVE_SQLITE3_STMT_READONLY
+    is_dml = ! sqlite3_stmt_readonly(st);
+    if (is_dml) {
+        /* Retain backwards-compatibility, as sqlite3_stmt_readonly will return
+         * false for BEGIN [IMMEDIATE|EXCLUSIVE] or DDL statements.
+         */
+        for (p = sql; *p != 0; p++) {
+            switch (*p) {
```

Comment:
I'd normally suggest avoiding repeating code here, but I think there's big chance that would make us to create an `#ifdef` nightmare :)

Reply:
Yes, that's why I chose to copy it.